base/bucket_gocb.go

```diff
@@ -1386,6 +1586,146 @@ func (bucket CouchbaseBucketGoCB) CouchbaseServerVersion() (major uint64, minor
 }
 
 func (bucket CouchbaseBucketGoCB) UUID() (string, error) {
+
+	// Temp workaround -- create a go-couchbase bucket just to get the UUID
 	// See https://github.com/couchbase/sync_gateway/issues/2418#issuecomment-289941131
-	return "error", fmt.Errorf("GoCB bucket does not expose UUID")
+	goCouchbaseBucket, err := GetCouchbaseBucket(bucket.spec, nil)
+	if err != nil {
+		return "", err
+	}
+
+	return goCouchbaseBucket.UUID()
+
```

Comment:
In the cbgt `Manager.CreateIndex()` method:

```
func (mgr *Manager) CreateIndex(sourceType,
	sourceName, sourceUUID, sourceParams,
	indexType, indexName, indexParams string, planParams PlanParams,
	prevIndexUUID string) error {
	atomic.AddUint64(&mgr.stats.TotCreateIndex, 1)

```

we pass the bucket uuid as the source uuid.  I think it does have to be the bucket uuid, because it uses it to check if the bucket uuid changes (if someone deletes and recreates a bucket with the same name).

I just filed https://issues.couchbase.com/browse/GOCBC-197